packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +71,66 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+
+function optionsToString(options: ScrollIntoViewOptions) {
+  return Object.keys(options).sort().map(key => `${key}${options[key]}`).toString();
+}
+
+export function scrollIntoViewFully(target: Element, scrollOptions?: ScrollIntoViewOptions) {
+  const intersectionObserverOptions: IntersectionObserverInit = {
+    root: undefined,
+    rootMargin: '0px',
+    // Safari doesn't work with theshold=1. Setting root to document.body or document.documentElement doesn't fix it either
+    threshold: isWebKit() ? .9 : 1
+  };
+
+  if (scrollOptions == null) {
+    scrollOptions = {
+      block: 'nearest'
+    };
+  }
+
+  let intersectionObserverCallback = (entries: Array<IntersectionObserverEntry>, observer: IntersectionObserver) => {
+    entries.forEach(entry => {
+      if (!entry.isIntersecting) {
+        scrollIntoViewHelper(entry.target, scrollOptions);
+      }
+      observer.unobserve(entry.target);
+    });
+  };
+
+  if (target) {
+    let observerMapKey = optionsToString(scrollOptions);
+    let observer = observerMap.get(observerMapKey);
+    if (!observer) {
+      observer = new IntersectionObserver(intersectionObserverCallback, intersectionObserverOptions);
+      observerMap.set(observerMapKey, observer);
+    }
+
+    observer.observe(target);
+  }
+}
+
+// TODO: rename? combine with scrollintoview above? Replace scrollIntoView above (would need to add param for scrollRef so that we could have old behavior)?
+// Perhaps add a parameter to this func to customize the root so we can say how far up we want to actually adjust the scroll?
+// scrollIntoView is exported and available from aria/utils so is it a breaking change to replace it with this func?
+// TODO: test with zoom/pinch zoom
+function scrollIntoViewHelper(target: Element, scrollOptions: ScrollIntoViewOptions) {
+  let root = document.scrollingElement || document.documentElement;
+  let scrollParent = getScrollParent(target);
+
+  // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+  if (!isScrollPrevented) {
+    target?.scrollIntoView?.(scrollOptions);
```

Comment:
Note to self and for posterity: I tried changing TableView [here](https://github.com/adobe/react-spectrum/blob/b35d5c02fe900badccd0cf1a8f23bb593419f238/packages/%40react-spectrum/table/src/TableView.tsx#L428) so that it used scrollIntoViewFully with the options of `{{block: center, inline: center}}` + a ref to the resizer div itself instead of the visually hidden input (which has problem with the IntersectionObserver) and it turned out a bit jankey. It worked better when the resizer was completely out of view since it would scroll more of the table into view thus giving the user more context but it would also continually shift the table's position when moving the resizer to past right edge of the table via keyboard.

Additionally there seems to be some behavior differences when doing `document.activeElement?.scrollIntoView?.({block: 'nearest', inline: 'nearest'});`  vs `scrollIntoViewFully(document.activeElement, {block: 'nearest', inline: 'nearest'});` even though they result in an basically identical call of `element.scrollIntoView` with the same args. Not entirely sure why, but will drop that in favor of the approach in https://github.com/adobe/react-spectrum/pull/3786/files instead